### ISSUE TITLE
Reach out to community repo / CNCF regarding 3rd party licenses

### ISSUE THREAD

**Issue by trask** [open-telemetry/opentelemetry-java-instrumentation#836]
The javaagent is different from many other projects because it embeds all of its dependencies in its distribution, so need to verify we are conforming to any CNCF requirements around 3rd party licenses that may be different for us (starting with https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md).

Also need to verify that we are conforming to Apache License requirement around NOTICE files that are contained in 3rd party dependencies that we embed.

Below is a listing of 3rd party dependencies we embed today and their licenses.

-------------------

```
./gradlew -q --no-parallel \
          :auto-tooling:dependencies --configuration runtimeClasspath \
          :auto-bootstrap:dependencies --configuration runtimeClasspath \
    | grep -- "---" | sed 's/.*--- //' | grep -v "^project " | grep -v "^io.opentelemetry:" | sed 's/\(.*\):[^:]*/\1/' | sort | uniq
```

shows these dependencies are included in `opentelemetry-javaagent.jar`:

```
com.blogspot.mydailyjava:weak-lock-free (Apache 2.0)
com.google.auto.service:auto-service (Apache 2.0)
com.google.auto:auto-common (Apache 2.0)
com.google.code.findbugs:jsr305 (Apache 2.0)
com.google.errorprone:error_prone_annotations (Apache 2.0)
com.google.guava:failureaccess (Apache 2.0)
com.google.guava:guava (Apache 2.0)
com.google.guava:listenablefuture (Apache 2.0)
com.google.j2objc:j2objc-annotations (Apache 2.0)
io.grpc:grpc-context (Apache 2.0)
net.bytebuddy:byte-buddy (Apache 2.0)
net.bytebuddy:byte-buddy-agent (Apache 2.0)
org.checkerframework:checker-compat-qual (MIT)
org.slf4j:slf4j-api (MIT)
org.slf4j:slf4j-simple (MIT)
```

while

```
./gradlew -q --no-parallel \
          :auto-exporters:opentelemetry-auto-exporter-otlp:dependencies --configuration runtimeClasspath \
          :auto-exporters:opentelemetry-auto-exporter-jaeger:dependencies --configuration runtimeClasspath \
          :auto-exporters:opentelemetry-auto-exporter-zipkin:dependencies --configuration runtimeClasspath \
          :auto-exporters:opentelemetry-auto-exporter-logging:dependencies --configuration runtimeClasspath \
    | grep -- "---" | sed 's/.*--- //' | grep -v "^project " | grep -v "^io.opentelemetry:" | sed 's/\(.*\):[^:]*/\1/' | sort | uniq
```

shows these additional dependencies are included in `opentelemetry-javaagent-all.jar`:

```
com.google.android:annotations (Apache 2.0)
com.google.api.grpc:proto-google-common-protos (Apache 2.0)
com.google.code.gson:gson (Apache 2.0)
com.google.protobuf:protobuf-java (BSD 3-clause)
com.google.protobuf:protobuf-java-util (BSD 3-clause)
com.squareup.okhttp3:okhttp (Apache 2.0)
com.squareup.okio:okio (Apache 2.0)
io.grpc:grpc-api (Apache 2.0)
io.grpc:grpc-core (Apache 2.0)
io.grpc:grpc-netty-shaded (Apache 2.0)
io.grpc:grpc-protobuf (Apache 2.0)
io.grpc:grpc-protobuf-lite (Apache 2.0)
io.grpc:grpc-stub (Apache 2.0)
io.opencensus:opencensus-api (Apache 2.0)
io.opencensus:opencensus-contrib-grpc-metrics (Apache 2.0)
io.perfmark:perfmark-api (Apache 2.0)
io.zipkin.reporter2:zipkin-reporter (Apache 2.0)
io.zipkin.reporter2:zipkin-sender-okhttp3 (Apache 2.0)
io.zipkin.reporter2:zipkin-sender-urlconnection (Apache 2.0)
io.zipkin.zipkin2:zipkin (Apache 2.0)
org.checkerframework:checker-compat-qual (MIT)
org.codehaus.mojo:animal-sniffer-annotations (MIT)
```

**Comment by trask** [comment]
Also see https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1165/files#r482757088

**Comment by trask** [comment]
Also see https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1189/files#diff-63095ccea8f6daf75d79dceb5d680acf and https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1189/files#diff-e994aad2ae74c3eb305aaaddbdbe0b82

**Comment by trask** [comment]
Also see https://github.com/cncf/foundation/blob/master/copyright-notices.md#what-about-third-party-code

**Comment by trask** [comment]
And https://github.com/open-telemetry/community/blob/master/CONTRIBUTING.md#copyright-notices

**Comment by iNikem** [comment]
@trask is there any conclusion for this task? Any recommendation how to handle these licenses? Or does it still require investigation?

**Comment by anuraaga** [comment]
For reference, we have default behavior from gradle-shadow-plugin. For shaded dependencies, we have an `inst/META-INF/maven` folder with one folder for each imported package containing the POM file for that package. For example https://repo1.maven.org/maven2/com/blogspot/mydailyjava/weak-lock-free/0.18/weak-lock-free-0.18.pom. 

In addition, we do seem to have "the first" copy of any file named NOTICE or NOTICE.txt. This means we are including the NOTICE for weak-lock-free, and the NOTICE.txt for commons-io, but nothing else.

Also random reference, for our distro, I use this Gradle plugin to generate license report

https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/build.gradle.kts#L25

Creates folder like this

https://github.com/aws-observability/aws-otel-java-instrumentation/tree/main/licenses

While we definitely need to confirm with CNCF, the bare minimum if we had no guidance that I would take is to do similar and bundle that `licenses` folder under META-INF.

**Comment by iNikem** [comment]
Closed by #2261
